py-polars/polars/dataframe/plotting.py

```diff
@@ -25,23 +26,29 @@
     else:
         from typing_extensions import Unpack
 
-    Encodings: TypeAlias = Dict[
-        str,
-        Union[X, Y, Color, Order, Size, Tooltip],
-    ]
+    Encoding: TypeAlias = Union[X, Y, Color, Order, Size, Tooltip]
+    Encodings: TypeAlias = Dict[str, Encoding]
+
+
+def _maybe_extract_shorthand(encoding: Encoding) -> Encoding:
+    if isinstance(encoding, alt.SchemaBase) and hasattr(encoding, "shorthand"):
```

Comment:
Will try to look at this properly tomorrow @MarcoGorelli 

This could be a source of false positives, you may be safer using `inspect.getattr_static` 
https://docs.python.org/3/library/inspect.html#fetching-attributes-statically

Reply:
thanks @dangotbanned !

the value does need returning, so maybe just using `getattr` directly is fine?